ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -182,6 +179,15 @@ impl<'a> Segment<'a> {
         self.x_left.len()
     }
 
+    /// This function returns the bit length of the `segment` in bits rounded up to a power of two.
+    ///
+    /// ## Panics
+    /// Panics when conversion from `u32` to `usize` fails.
+    #[must_use]
+    pub fn power_of_two_bit_len(&self) -> usize {
```

Comment:
There is a function [`next_power_of_two`](https://doc.rust-lang.org/std/primitive.usize.html#method.next_power_of_two) on the built-in integer types, I'm not sure the `power_of_two_bit_len` functions are worthwhile over `foo.len().next_power_of_two()`.